network/src/main/kotlin/com/bitwarden/network/interceptor/HeadersInterceptor.kt

```diff
@@ -0,0 +1,28 @@
+package com.bitwarden.network.interceptor
+
+import com.bitwarden.network.util.HEADER_KEY_CLIENT_NAME
+import com.bitwarden.network.util.HEADER_KEY_CLIENT_VERSION
+import com.bitwarden.network.util.HEADER_KEY_DEVICE_TYPE
+import com.bitwarden.network.util.HEADER_KEY_USER_AGENT
+import com.bitwarden.network.util.HEADER_VALUE_DEVICE_TYPE
+import okhttp3.Interceptor
+import okhttp3.Response
+
+/**
+ * Interceptor responsible for adding various headers to all API requests.
+ */
+class HeadersInterceptor(
+    private val userAgent: String,
+    private val clientName: String,
+    private val clientVersion: String,
```

Comment:
🗒️ We are not using accessors because these values are static so we only need clients to provide them once during instantiation. `@Provides` functions have been updated to use the client `BuildConfig` to provide the arguments.